### PR TITLE
WIP: Add a 'enabled' field for triggers

### PR DESCRIPTION
**Pull request by lakshmi-kannan**

### CODE REVIEW

File: st2common/st2common/models/api/rule.py [hunk 1]
```diff
@@ -239,7 +239,8 @@ def to_model(cls, rule):
 
         # Create a trigger for the provided rule
         trigger_db = TriggerService.create_trigger_db_from_rule(rule)
-        kwargs['trigger'] = reference.get_str_resource_ref_from_model(trigger_db)
+        if trigger_db:
```
Comment (Kami) [inline st2common/st2common/models/api/rule.py]: Just curious why this change is needed.

File: st2common/st2common/models/db/trigger.py [hunk 2]
```diff
@@ -77,6 +77,8 @@ class TriggerDB(stormbase.StormBaseDB, stormbase.ContentPackResourceMixin,
 
     name = me.StringField(required=True)
     pack = me.StringField(required=True, unique_with='name')
+    enabled = me.BooleanField(required=True, default=True,
```
Comment (Kami) [inline st2common/st2common/models/db/trigger.py]: I know it's WIP, but before we merge it we should also document what "disabled trigger" implies (what kind of behavior, etc.).

**Comment by Kami** [comment]
^ Bump let's get that finished.

**Comment by lakshmi-kannan** [comment]
This PR is not needed anymore.
